Proxy.h

```diff
@@ -54,15 +56,21 @@ class ProxyTracker
 	};
 
 	LinkedProxy* head;
-	pthread_mutex_t lock;
+	std::mutex lock;
 };
 
 template <class RefAllocator, class ...TX>
 class ProxyGenerator : public ProxyObject, public TX::__Proxy...
 {	
 protected:
-	ProxyGenerator() : m_ProxyObject(NewInstance(this, (jobject[]){TX::__CLASS...}, sizeof...(TX)))	
+	ProxyGenerator() 
+#if WINDOWS
+	: m_ProxyObject(NewInstance(this, TX::__CLASS...))
 	{
+#else
+	: m_ProxyObject(NewInstance(this, (jobject[]){TX::__CLASS...}, sizeof...(TX)))	
+	{
+#endif
```

Comment:
Can you at least create a task to clean this up. I really don't like having separate code paths where can be only one.
In this case you can pass additional nullptr to NewInstance at the end and have it figure out the number of interfaces via va_list.

Reply:
Agreed, @aurimasc I can actually remove #else block and leave it how it's written for windows.

It seems to compile fine - https://github.com/Unity-Technologies/android-jni-bridge/pull/36/commits/d63fa4b827907520f1fef8e4e7cf3d848f90ffca